src/core/js/models/adaptModel.js

```diff
@@ -1,859 +1,853 @@
-          this.triggerTrackableState.isQueued = false;
+        // Flag that the function is awaiting trigger
+        this.triggerTrackableState.isQueued = true;
 
-        }, 17)
-      );
+      },
+      _.debounce(() => {
 
-      // Listen to model changes, trigger trackable state change when appropriate
-      this.listenTo(this, 'change', ({ changed }) => {
+        // Trigger original function
+        originalTrackableStateFunction.apply(this);
 
-        // Skip if trigger queued or adapt hasn't started yet
-        if (this.triggerTrackableState.isQueued || !Adapt.attributes._isStarted) {
-          return;
-        }
+        // Unset waiting flag
+        this.triggerTrackableState.isQueued = false;
 
-        // Check that property is trackable
-        const trackablePropertyNames = _.result(this, 'trackable', []);
-        const changedPropertyNames = Object.keys(changed);
-        const isTrackable = changedPropertyNames.find(item => {
-          return trackablePropertyNames.includes(item);
-        });
+      }, 17)
+    );
 
-        if (isTrackable) {
-          // Trigger trackable state change
-          this.triggerTrackableState();
-        }
-      });
-    }
+    // Listen to model changes, trigger trackable state change when appropriate
+    this.listenTo(this, 'change', ({ changed }) => {
 
-    setupChildListeners() {
-      const children = this.getChildren();
-      if (!children.length) {
+      // Skip if trigger queued or adapt hasn't started yet
+      if (this.triggerTrackableState.isQueued || !Adapt.attributes._isStarted) {
         return;
       }
 
-      this.listenTo(children, {
-        'all': this.onAll,
-        'bubble': this.bubble,
-        'change:_isReady': this.checkReadyStatus,
-        'change:_isComplete': this.onIsComplete,
-        'change:_isInteractionComplete': this.checkInteractionCompletionStatus
+      // Check that property is trackable
+      const trackablePropertyNames = _.result(this, 'trackable', []);
+      const changedPropertyNames = Object.keys(changed);
+      const isTrackable = changedPropertyNames.find(item => {
+        return trackablePropertyNames.includes(item);
       });
-    }
 
-    init() {}
-
-    getTrackableState() {
+      if (isTrackable) {
+        // Trigger trackable state change
+        this.triggerTrackableState();
+      }
+    });
+  }
 
-      const trackable = this.resultExtend('trackable', []);
-      const json = this.toJSON();
+  setupChildListeners() {
+    const children = this.getChildren();
+    if (!children.length) {
+      return;
+    }
 
-      const args = trackable;
-      args.unshift(json);
+    this.listenTo(children, {
+      'all': this.onAll,
+      'bubble': this.bubble,
+      'change:_isReady': this.checkReadyStatus,
+      'change:_isComplete': this.onIsComplete,
+      'change:_isInteractionComplete': this.checkInteractionCompletionStatus
+    });
+  }
 
-      return _.pick(...args);
+  init() {}
```

Comment:
I think there might need to be a space between the semicolons? `init() { }`

Reply:
It seems fine in the linting and compiling. It's one of those functions designed for override.